File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/cynen/sellersgoods/service/BrandService.java
package com.cynen.sellersgoods.service;

import java.util.List;
import java.util.Map;

import com.cynen.pojo.TbBrand;

import entity.PageResult;
import entity.Result;

public interface BrandService {

	/**
	 * 查询全部品牌.
	 * @return
	 */
	public List<TbBrand> findAll();
	
	// 分页查询
	public PageResult findPage(int currPage,int pageSize);
	// 添加品牌
	public void addbrand(TbBrand tbBrand);
	
	public void updatebrand(TbBrand tbBrand);
	
	// 删除
	public void deleteBrand(Long[] ids);
	
	// 条件分页查询
	public PageResult findPage(TbBrand tbBrand,int currPage,int pageSize);
	
	// 查询一个
	public TbBrand findOne(Long id);
	
	// 查询所有品牌列表,select2使用.
	List<Map> selectOptionList();
}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
	// 品牌服务.
	app.service('brandService',function($http){
		this.findAll=function(){
			return $http.get("../brand/findAll.do");
		}
		
		this.findPage=function(currPage,pageSize){
			return $http.post("../brand/findPage.do?currPage="+currPage+"&pageSize="+pageSize);
		}
		
		this.delte=function(selectIds){
			return $http.get("../brand/delte.do?ids="+selectIds);
		}
		
		this.search = function(currPage,pageSize,searchEntity){
			return $http.post("../brand/search.do?currPage="+currPage+"&pageSize="+pageSize,searchEntity);
		}
		
		this.add=function(entity){
			return $http.post("../brand/add.do",entity);
		}
		this.update=function(entity){
			return $http.post("../brand/update.do",entity);
		}
		
		this.findOne=function(id){
			return $http.get("../brand/findOne.do?id="+id);
		}
	})